cmd/bucket-notification-handlers.go

```diff
@@ -145,10 +145,14 @@ func (api objectAPIHandlers) PutBucketNotificationHandler(w http.ResponseWriter,
 	}
 
 	rulesMap := config.ToRulesMap()
-	globalNotificationSys.AddRulesMap(bucketName, rulesMap)
-	for nerr := range globalNotificationSys.PutBucketNotification(bucketName, rulesMap) {
-		logger.GetReqInfo(ctx).AppendTags("remotePeer", nerr.Host.Name)
-		logger.LogIf(ctx, nerr.Err)
+	if len(rulesMap) == 0 {
+		globalNotificationSys.RemoveNotification(bucketName)
```

Comment:
This removes all notification configuration including listeners.  This won't work.

The issue is at https://github.com/minio/minio/blob/master/cmd/notification.go#L383

Reply:
An empty rulesMap actually indicates to remove all the notifications right ? @balamurugana

Reply:
Yep, but not applicable to bucket listeners

Reply:
Okay ! Got it

Reply:
Made the changes @balamurugana . Please check